tests/unit/test_provider.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+import logging
+from pathlib import Path
+from unittest.mock import PropertyMock, patch
+
+import pytest
+import yaml
+from ops.testing import Harness
+
+from charm import KafkaCharm
+from literals import CHARM_KEY, PEER, REL_NAME
+
+logger = logging.getLogger(__name__)
+
+CONFIG = str(yaml.safe_load(Path("./config.yaml").read_text()))
+ACTIONS = str(yaml.safe_load(Path("./actions.yaml").read_text()))
+METADATA = str(yaml.safe_load(Path("./metadata.yaml").read_text()))
+
+
+@pytest.fixture
+def harness():
+    harness = Harness(KafkaCharm, meta=METADATA)
+    harness.add_relation("restart", CHARM_KEY)
+    harness._update_config(
+        {
+            "offsets-retention-minutes": 10080,
+            "log-retention-hours": 168,
+            "auto-create-topics": False,
+        }
+    )
+
+    harness.begin()
+    return harness
+
+
+def test_client_relation_created_defers_if_not_ready(harness):
```

Comment:
Adding clear docstrings to your unit tests would be super helpful 🙏

Reply:
This is a personal difference, but do the test-names not document sufficiently?

Reply:
Or, like your main comment, are you suggesting they be added to give context?

Reply:
I personally think it's one way or the other, concise test_method names + docstring or descriptive enough test method names and no docstrings 
For example: [here](https://github.com/canonical/tls-certificates-operator/blob/main/tests/unit/test_charm.py#L54)) - they follow the convention: `given_x_when_y_then_z` where: `x: Initial situation, y: Action (ideally the method that is under test), z: Expected behaviour`